library/core/src/sync/atomic.rs

```diff
@@ -211,6 +211,8 @@
 #![cfg_attr(not(target_has_atomic_load_store = "8"), allow(dead_code))]
 #![cfg_attr(not(target_has_atomic_load_store = "8"), allow(unused_imports))]
 #![rustc_diagnostic_item = "atomic_mod"]
+// AtomicPtr intrinsics are unsafe and take pointers, but do not dereference them.
+#![allow(clippy::not_unsafe_ptr_arg_deref)]
```

Comment:
Don't these call intrinsics which are themselves unsafe functions?

Reply:
Yes, they are unsafe, which is what clippy is concerned about. I elaborated the comment to explain what exactly is happening.